architecture-standards/0005-introduce-core-ui-architecture-module.md

```diff
@@ -0,0 +1,34 @@
+# ADR-0004 - Introduce a UI architecture module to the core platform
+
+## Date
+
+2024-02-07
+
+## Context
+
+The Gradle core platform provides many services to the Gradle platforms and build logic. One such group of services allow logic to interact with the build user, to provide diagnostics, progress information, prompt for questions, and so on. Currently, these services are part of the core platform runtime architecture module.
+
+A downside of this structure is that it is difficult to do focussed work on the Gradle UI.
+
+## Decision
+
+Introduce a "UI" architecture module to the core platform, and move the user interaction services to this new module.
```

Comment:
❓ Where will this module live in the dependency graph of Core architecture modules?  What is the final proposed dependency graph of those modules?

Reply:
```
core/runtime <-- core/ui 
                 core/execution <-- (configures) core/configuration
```